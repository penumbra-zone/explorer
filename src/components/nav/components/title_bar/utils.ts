export const formatMarket = (data: {
  marketCap: number;
  communityPool: number;
  price: number;
  inflation: number;
}) => {
  return ([
    {
      key: 'price',
      data: data.price,
    },
    {
      key: 'marketCap',
      data: data.marketCap,
    },
    {
      key: 'inflation',
      data: data.inflation,
    },
    {
      key: 'communityPool',
      data: data.communityPool,
    },
  ]);
};
