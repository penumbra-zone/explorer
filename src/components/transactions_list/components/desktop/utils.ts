export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'slot',
    width: 15,
  },
  {
    key: 'signature',
    width: 30,
  },
  {
    key: 'amount',
    align: 'right',
    width: 20,
  },
  {
    key: 'result',
    align: 'right',
    width: 15,
  },
  {
    key: 'time',
    align: 'right',
    width: 20,
  },
];
