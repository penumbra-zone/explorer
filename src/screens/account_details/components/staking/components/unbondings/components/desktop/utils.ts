export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 25,
  },
  {
    key: 'commission',
    align: 'right',
    width: 25,
  },
  {
    key: 'amount',
    width: 25,
    align: 'right',
  },
  {
    key: 'linkedUntil',
    width: 25,
    align: 'right',
  },
];
