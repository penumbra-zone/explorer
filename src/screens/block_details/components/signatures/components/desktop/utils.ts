export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 34,
  },
  {
    key: 'votingPower',
    align: 'right',
    width: 33,
  },
  {
    key: 'signed',
    width: 33,
  },
];
