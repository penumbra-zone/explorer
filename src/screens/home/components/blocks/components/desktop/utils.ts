export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
}[] = [
  {
    key: 'height',
  },
  {
    key: 'validator',
  },
  {
    key: 'hash',
  },
  {
    key: 'txs',
  },
  {
    key: 'time',
    align: 'right',
  },
];
