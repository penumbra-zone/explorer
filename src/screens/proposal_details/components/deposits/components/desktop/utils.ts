export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'depositor',
    width: 34,
  },
  {
    key: 'amount',
    width: 33,
    align: 'right',
  },
  {
    key: 'time',
    width: 33,
  },
];
