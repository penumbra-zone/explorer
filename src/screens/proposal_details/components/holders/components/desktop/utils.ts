export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 10,
  },
  {
    key: 'address',
    width: 30,
  },
  {
    key: 'quantity',
    width: 20,
    align: 'right',
  },
  {
    key: 'percentage',
    width: 20,
  },
  {
    key: 'value',
    align: 'right',
    width: 20,
  },
];
