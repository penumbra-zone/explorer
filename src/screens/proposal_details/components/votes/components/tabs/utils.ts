export const tabLabels = [
  {
    key: 'all',
    num: 30,
  },
  {
    key: 'yes',
    num: 3,
  },
  {
    key: 'no',
    num: 3,
  },
  {
    key: 'noWithVeto',
    num: 3,
  },
];
