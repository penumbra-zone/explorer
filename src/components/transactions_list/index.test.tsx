import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import TransactionsList from '.';

// ==================================
// mocks
// ==================================
jest.mock('./components', () => ({
  Mobile: (props) => <div id="Mobile" {...props} />,
  Desktop: (props) => <div id="Desktop" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('components: TransactionsList', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <TransactionsList
          items={[
            {
              block: '812,768,640',
              hash: '76nwV8zz8tLz97SBRXH6uwHvgHXtqJDLQfF66jZhQ857',
              messages: 123,
              success: false,
              time: 1615187146246,
            },
          ]}
          itemCount={1}
        />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
