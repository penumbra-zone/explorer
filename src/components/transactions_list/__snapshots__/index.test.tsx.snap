// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: TransactionsList matches snapshot 1`] = `
<div
  className=""
>
  <div
    className="makeStyles-mobile"
    id="Mobile"
    useContext={[MockFunction]}
  />
  <div
    className="makeStyles-desktop"
    id="Desktop"
    useContext={[MockFunction]}
  />
</div>
`;
